pkg/px-resource-gateway/px_resource_gateway.go

```diff
@@ -0,0 +1,84 @@
+package server
+
+import (
+	"context"
+
+	pb "github.com/libopenstorage/operator/proto"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/sirupsen/logrus"
+	emptypb "google.golang.org/protobuf/types/known/emptypb"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type server struct {
+	k8s         core.Ops
+	semaphorePQ SemaphorePriorityQueue
+	pb.UnimplementedSemaphoreServiceServer
+}
+
+// TODO implement semaphore map for different resource ids
+func NewServer() *server {
+	s := &server{
+		k8s:         core.Instance(),
+		semaphorePQ: NewSemaphorePriorityQueue(),
+	}
+	return s
+}
+
+func (s *server) AcquireLock(ctx context.Context, req *pb.AcquireLockRequest) (*pb.AcquireLockResponse, error) {
+	// validate request
+	if req.ResourceId == "" {
+		return &pb.AcquireLockResponse{}, status.Error(codes.InvalidArgument, "Resource ID is required")
+	}
+	if req.ClientId == "" {
+		return &pb.AcquireLockResponse{}, status.Error(codes.InvalidArgument, "Client ID is required")
+	}
+	accessPriority := req.AccessPriority
+	if accessPriority == pb.SemaphoreAccessPriority_TYPE_UNSPECIFIED {
```

Comment:
I would highly discourage default values for priority. Make the client be clear and if they dont pass a priority return an error that the priority must be set